core/src/main/scripts/importer/importOncokbDiscreteCNA.py

```diff
@@ -0,0 +1,288 @@
+    return list(elements.values())
+
+
+def update_cna_metafile(meta_cna_file_path, pd_file_name):
+    """Add reference to pd annotation file to CNA meta file."""
+    meta_file = open(meta_cna_file_path, "r")
+    lines = meta_file.readlines()
+    meta_file.close()
+    meta_file_name = os.path.basename(meta_cna_file_path)
+    dir = os.path.dirname(meta_cna_file_path)
+    os.rename(meta_cna_file_path, os.path.join(dir, 'ONCOKB_IMPORT_BACKUP_' + meta_file_name))
+    if lines[-1] == '\n':
+        lines = lines[:-1]
+    meta_file = open(meta_cna_file_path, "w")
+    for line in lines:
+        meta_file.write(line)
+    meta_file.write('pd_annotations_filename: ' + pd_file_name + '\n')
+    meta_file.close()
+
+
+def write_annotations_to_file(features, pd_file_name):
+    """Write CNA pd annotations to data file."""
+    new_file = open(pd_file_name, "w")
+    new_file.write("SAMPLE_ID\tEntrez_Gene_Id\tcbp_driver\tcbp_driver_annotation\tcbp_driver_tiers\tcbp_driver_tiers_annotation\n")
+    for feature in features:
+        if 'oncogenic' in feature:
+            line = "\t".join(
+                [feature['sample_id'], feature['Entrez_Gene_Id'], libImportOncokb.evaluate_driver_passenger(feature['oncogenic']),
+                feature['oncogenic'], '', '']) + "\n"
+            new_file.write(line)
+    new_file.close()
+
+
+def check_required_columns(header_elements):
+    missing_columns = []
+    for required_column in required_cna_columns:
+        if not required_column in header_elements:
+            missing_columns.append(required_column)
+    if len(missing_columns) > 0:
+        raise RuntimeError("One or more required columns for OncoKb import are missing from the discrete CNA file. " \
+                           "Missing column(s): [" + ", ".join(missing_columns) + "]")
+
+
+def interface():
+    parser = argparse.ArgumentParser(description='cBioPortal OncoKB annotation importer')
+    parser.add_argument('-u', '--url_server',
+                        type=str,
+                        default='http://localhost:8080',
+                        help='URL to cBioPortal server. You can '
+                             'set this if your URL is not '
+                             'http://localhost:8080')
+    parser.add_argument('-m', '--study_directory', type=str, required=True,
+                        help='path to study directory.')
+    parser = parser.parse_args()
+    return parser
+
+
+if __name__ == '__main__':
+    try:
+        parsed_args = interface()
+        exit_status = main_import(parsed_args)
+    finally:
+        logging.shutdown()
+        del logging._handlerList[:]  # workaround for harmless exceptions on exit
+    print(('Import of OncoKB annotations for discrete CNA {status}.'.format(
+        status={0: 'succeeded',
+                1: 'failed',
+                2: 'not performed as problems occurred',
+                3: 'succeeded with warnings'}.get(exit_status, 'unknown'))), file=sys.stderr)
```

Comment:
Eh, where is it duplicated? You mean between the CNA and mutation importer files?

Reply:
@pvannierop yeah, my main concern for this pr is the duplication between CNA and mutation files. But it could be too much trouble to share the logics between two. Let me know what you think.

Reply:
Well, i agree that every script has some boilerplate logic that it needs to check arguments, setup logging etc. But what is it that you propose exactly? Do you think it is better to merge the mutation and CNA import scripts into a single script so that the code redundancy is removed? I personally like the fact that the logic is in different files. I have moved the _interface()_ method to the lib file so that both scripts share this code. Is that Ok with you?

Reply:
My concern is not about the check arguments/setup logging but rather overall code duplications. But I think both you and Oleguer have updated this pr to fix that.

Reply:
Ah yes, we have created a lib module that both scripts use.